modules/core/03-connection/keeper/handshake.go

```diff
@@ -288,11 +288,16 @@ func (k Keeper) ConnOpenConfirm(
 	return nil
 }
 
-func (k Keeper) validateHeight(
+// validateConsensusHeight validates the consensus height the counterparty is using to store a representation
+// of this chain's consensus state. For non-localhost clients, the consensus height must be a past height.
+// The localhost consensus height must be less than or equal to the current height.
+func (k Keeper) validateConsensusHeight(
 	ctx sdk.Context,
 	clientType string,
 	consensusHeight exported.Height,
 ) error {
+	// if the consensusHeight is greater than the current height,
+	// the self consensus state will fail to be retrieved.
 	if clientType == exported.Localhost {
```

Comment:
I think I'd actually prefer to check the height still in this case. It isn't necessary, but I think it is good to avoid skipping checks altogether for the localhost, it makes more sense to apply the logic of the check even if it is adjust just slightly (in this case, the current height is a valid consensus height). How would people feel about me adjust this code?

Reply:
Sounds good to me

Reply:
If we update the connection handshake to disallow working for the localhost client, we can remove this entirely?

Reply:
Yeah. We can revert the addition of this `validateHeight` function back to the original code if you like!

Reply:
Yea I think I'll revert the changes just because this check is a bit naunced and not reusable after these two checks